File: 49-group-anagrams/group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    map = {}
    for (let str of strs) {
        let sortedStr = [...str].sort().join('')
        if (map[sortedStr]) map[sortedStr].push(str)
        else map[sortedStr] = [str]
    }
    let ans = Object.values(map)
    return ans
}

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:

        hash_map = {}
        

        for s in strs:
            key = "".join(sorted(s))
            
            #print(hash_map)
            if(key in hash_map):
                hash_map[key].append(s)
            else:
                hash_map[key] = [s]

        return list(hash_map.values())
             


        

            
        

             
        